pint/models/model_builder.py

```diff
@@ -70,8 +70,15 @@ def preprocess_parfile(self, parfile):
         """
         param = {}
         repeat_par = {}
-        pfile = open(parfile, 'r')
-        for l in [pl.strip() for pl in pfile.readlines()]:
+        if isinstance(parfile, str):
+            pfile = open(parfile, 'r')
+            plines = pfile.readlines()
+        elif isinstance(parfile, (list, tuple)):
```

Comment:
Lists and tuples are not the only iterable types. Perhaps astropy's isiterable, or just try doing a for loop over it and catch the exception?

Reply:
> I think the ability to easily run F tests is very useful, as is the ability to take par files apart and put them together differently, but wouldn't it be simpler to just operate by freezing one or more parameters at their best values?

That would be easier. However, since the model (e.g., the binary model) is selected dependent on the parameters that are given, rebuilding the model ensures that PINT uses the proper model.

Reply:
Well, that's sort of the problem: Because model selection is so opaque, how can you be sure the F test is valid if you rebuilt your TimingModel from scratch? If the model with the new parameters doesn't specialize to the one you started with when you fix certain parameters, the F test will give nonsense results. If the larger model does specialize correctly, then freezing the parameters will give you exactly what you expect.

If you're unsure whether an astrometry model that permits proper motion but with the proper motion fixed to zero is the same as a model with no proper motion, then you had better not apply the F test!